src/meltano/core/project.py

```diff
@@ -416,122 +423,157 @@ def dotenv_update(self):
         self.refresh()
 
     @makedirs
-    def meltano_dir(self, *joinpaths):
+    def meltano_dir(self, *joinpaths: StrPath, make_dirs: bool = True) -> Path:  # noqa: ARG002
```

Comment:
The `make_dirs` parameter here and in some other functions appears unused. If it isn't supported but is included for the signature, or if it must be `True`, then a `NotImplementedError` or `ValueErrror` exception should be raised otherwise.

Reply:
It's actually required so we can pass down the flag from methods like `Project.logs_dir`. For example, without it the following code fails

```python
from meltano.core.project import Project

project = Project.find(".")
project.logs_dir(make_dirs=False)
```

with

```python
TypeError: Project.meltano_dir() got an unexpected keyword argument 'make_dirs'
```

That's because all functions decorated with `@makedirs` are expected to accept the `make_dirs` keyword argument after the changes introduced to the decorator in this PR.

Wdyt?

Reply:
After reading through the code more closely, and understanding it better, my impression is that the `makedirs` decorator is a bad idea. Directory creation shouldn't be inherently tied to retrieving information about what those directories would be. That said, changing that goes beyond the scope of this PR, which provides a valuable fix.